consensus/state.go

```diff
@@ -1306,6 +1306,22 @@ func (cs *ConsensusState) addProposalBlockPart(height int64, part *types.Part, v
 		}
 		// NOTE: it's possible to receive complete proposal blocks for future rounds without having the proposal
 		cs.Logger.Info("Received complete proposal block", "height", cs.ProposalBlock.Height, "hash", cs.ProposalBlock.Hash())
+
+		// Update ValidBlock
+		prevotes := cs.Votes.Prevotes(cs.Round)
+		blockID, ok := prevotes.TwoThirdsMajority()
+		if ok && !blockID.IsZero() && (cs.ValidRound < cs.Round) {
+			// update valid value
+			if !cs.ValidBlock.HashesTo(blockID.Hash) && cs.ProposalBlock.HashesTo(blockID.Hash) {
```

Comment:
See comment below, I believe we want this if condition to simply be `if cs.ProposalBlock.HashesTo(blockID.Hash) {`